### ISSUE TITLE
What's the relationship with https://github.com/ocharles/haskell-opentracing

### ISSUE THREAD

**Issue by CMCDragonkai** [kim/opentracing#10]
Is there any collaboration between these 2 projects?

**Comment by kim** [comment]
Both projects started out as an experiment using GHC's [Backpack](https://ghc.haskell.org/trac/ghc/wiki/Backpack) feature. As I ran into some issues, I decided to move away from it and get a library which is usable where Backpack is not supported (eg. stack). I had a brief conversation with @ocharles, but since I moved away from Backpack, no further collaboration arose. I'm not sure also whether @ocharles' version is still actively developed, whereas this library is about to be put into production (still quite some polishing to be done before it will land on hackage, though).

**Comment by NicolasT** [comment]
Shameless plug: whilst not really about OpenTracing, at some point I also started https://github.com/haskell-service/jaeger
Some convergence might be... beneficial.

**Comment by kim** [comment]
@CMCDragonkai Yeah I've been following that, but wasn't sure if it entails to be a member of the OTSC, CNCF, or somesuch. So far, I found it to be not so easy to translate the spec to something halfway idiomatic in Haskell. I think I arrived at something pretty close to other languages now, while still usable in "mtl-style" stacks. Other paradigms may be built on top.

If you have any thoughts on the API side of things, please let me know!

**Comment by ocharles** [comment]
> I had a brief conversation with @ocharles, but since I moved away from Backpack, no further collaboration arose. I'm not sure also whether @ocharles' version is still actively developed, whereas this library is about to be put into production (still quite some polishing to be done before it will land on hackage, though).

Yep, mine stuck as an experiment in Backpack, and the fizzled out. Congratulations on the near release! I'd love to add tracing to some work stuff, so I will hopefully get a chance to play with this in the future.
